sorts/odd_even_transposition_parallel.py

```diff
@@ -72,6 +72,34 @@ def oe_process(position, value, l_send, r_send, lr_cv, rr_cv, result_pipe):
 
 
 def odd_even_transposition(arr):
+    """
+    >>> odd_even_transposition(list(range(11)[::-1])) == sorted(list(range(11)[::-1]))
+    True
+    >>> odd_even_transposition(["a", "x", "c"]) == sorted(["x", "a", "c"])
+    True
+    >>> odd_even_transposition([1.9, 42.0, 2.8]) == sorted([1.9, 42.0, 2.8])
+    True
+    >>> odd_even_transposition([False, True, False]) == sorted([False, False, True])
+    True
+    >>> odd_even_transposition([1, 32.0, 9]) == sorted([False, False, True])
+    False
+    >>> odd_even_transposition([1, 32.0, 9]) == sorted([1.0, 32, 9.0])
+    True
+    >>> unsorted_list = [-442, -98, -554, 266, -491, 985, -53, -529, 82, -429]
+    >>> odd_even_transposition(unsorted_list) == sorted(unsorted_list)
+    True
+    >>> unsorted_list = [-442, -98, -554, 266, -491, 985, -53, -529, 82, -429]
+    >>> odd_even_transposition(unsorted_list) == sorted(unsorted_list + [1])
+    False
+    >>> odd_even_transposition([False, "a", 8]) == sorted([False, "a", 8])
+    Traceback (most recent call last):
+        ...
+    TypeError: '>' not supported between instances of 'bool' and 'str'
+    >>> odd_even_transposition([8, "a"]) == sorted(["a", 8])
+    Traceback (most recent call last):
+        ...
+    TypeError: '<' not supported between instances of 'str' and 'int'
```

Comment:
```suggestion
    # >>> odd_even_transposition(list(range(11)[::-1])) == sorted(list(range(11)[::-1]))
    True
    # >>> odd_even_transposition(["a", "x", "c"]) == sorted(["x", "a", "c"])
    True
    # >>> odd_even_transposition([1.9, 42.0, 2.8]) == sorted([1.9, 42.0, 2.8])
    True
    # >>> odd_even_transposition([False, True, False]) == sorted([False, False, True])
    True
    # >>> odd_even_transposition([1, 32.0, 9]) == sorted([False, False, True])
    False
    # >>> odd_even_transposition([1, 32.0, 9]) == sorted([1.0, 32, 9.0])
    True
    >>> unsorted_list = [-442, -98, -554, 266, -491, 985, -53, -529, 82, -429]
    # >>> odd_even_transposition(unsorted_list) == sorted(unsorted_list)
    True
    >>> unsorted_list = [-442, -98, -554, 266, -491, 985, -53, -529, 82, -429]
    # >>> odd_even_transposition(unsorted_list) == sorted(unsorted_list + [1])
    False
    # >>> odd_even_transposition([False, "a", 8]) == sorted([False, "a", 8])
    Traceback (most recent call last):
        ...
    TypeError: '>' not supported between instances of 'bool' and 'str'
    # >>> odd_even_transposition([8, "a"]) == sorted(["a", 8])
    Traceback (most recent call last):
        ...
    TypeError: '<' not supported between instances of 'str' and 'int'
```

Reply:
Just finished pushing all the commits to experiment with. As expected, commenting out all the tests allowed the build to pass. I've made one commit for each test, leaving the rest of the tests commented out except that one. The only exception was the last 3, which were expected to throw exceptions as I couldn't even get those to work locally. If they all fail to build I'm not sure how to proceed.

Reply:
OK.  Please uncomment all the tests that work so that they run.  Then **leave a blank line** and then the commented out tests that seem to run forever.  I will look at those in my morning.  Thanks for your persistence!!

Reply:
That won't be necessary as each test failed individually! I'm not familiar enough with the way Python does parallelization but I suspect that has something to do with why these tests all fail.

Reply:
After a lot of trial and error I finally figured it out! It had to do with the globally defined process lock. The only thing I couldn't figure out was how to test for multiple data types. It threw out multiple exceptions with each of those tests and I couldn't figure out how to get those working in doctest. I tried adding ```# doctest: +IGNORE_EXCEPTION_DETAIL``` as well as playing around with ellipses (```...```) to no avail.

Sample output:
```
TypeError: '<' not supported between instances of 'str' and 'bool'
TypeError: '>' not supported between instances of 'bool' and 'str'
```